# Cognifyz Internship -Level 1
# Task 2- Descriptive Analysis
Calculate basic statistical measures (mean,median, standard deviation, etc.) for numericalcolumns.
Explore the distribution of categoricalvariables like "Country Code," "City," and"Cuisines."
Identify the top cuisines and cities with thehighest number of restaurants.

## 1.Importing necessary libararies

In [277]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## 2. Loading dataset

In [280]:
df=pd.read_csv("data.csv")
df.head()

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)               Yes   
1                          Japanese  ...  Botswana Pula(P)               Yes   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)               Yes   
3                   Japanese, Sushi  ...  Botswana Pula(P)                No   
4                  Japanese, Korean  ...  Botswana Pula(P)               Yes   

  Has Online delivery Is delivering now Switch to order menu Price range  \
0                  No                No                   No           3   
1                  No                No                   No           3   
2                  No                No                   No           4   
3                  No                No                   No           4   
4                  No                No                   No           4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  
3               4.9    Dark Green   Excellent   365  
4               4.8    Dark Green   Excellent   229  

[5 rows x 21 columns]

In [282]:
# Removing special characters
for col in df.select_dtypes(include='object'):
    df[col] = df[col].str.replace(r'[^a-zA-Z0-9, ]+', '', regex=True)

## 3.Statistical summary of numerical fields

In [285]:

df.drop(columns=['Restaurant ID','Country Code','Longitude','Latitude']).describe(include='number')

Average Cost for two  Price range  Aggregate rating         Votes
count           9551.000000  9551.000000       9551.000000   9551.000000
mean            1199.210763     1.804837          2.666370    156.909748
std            16121.183073     0.905609          1.516378    430.169145
min                0.000000     1.000000          0.000000      0.000000
25%              250.000000     1.000000          2.500000      5.000000
50%              400.000000     2.000000          3.200000     31.000000
75%              700.000000     2.000000          3.700000    131.000000
max           800000.000000     4.000000          4.900000  10934.000000

In [287]:
df.drop(columns=['Restaurant ID','Country Code','Longitude','Latitude']).median(numeric_only=True)


Average Cost for two    400.0
Price range               2.0
Aggregate rating          3.2
Votes                    31.0
dtype: float64

## 4. Distribution of categorical variables

## Country Code

In [291]:
df['Country Code'].value_counts()

Country Code
1      8652
216     434
215      80
30       60
214      60
189      60
148      40
208      34
14       24
162      22
94       21
184      20
166      20
191      20
37        4
Name: count, dtype: int64

In [293]:
df['Country Code'].nunique()

15

Comment :
- Most of the restaurants are in country code 1.
- There are 15 unique countries.

## City

In [297]:
df['City'].unique()

array(['Makati City', 'Mandaluyong City', 'Pasay City', 'Pasig City',
       'Quezon City', 'San Juan City', 'Santa Rosa', 'Tagaytay City',
       'Taguig City', 'Braslia', 'Rio de Janeiro', 'So Paulo', 'Albany',
       'Armidale', 'Athens', 'Augusta', 'Balingup', 'Beechworth', 'Boise',
       'Cedar RapidsIowa City', 'ChathamKent', 'Clatskanie', 'Cochrane',
       'Columbus', 'Consort', 'Dalton', 'Davenport', 'Des Moines',
       'Dicky Beach', 'Dubuque', 'East Ballina', 'Fernley', 'Flaxton',
       'Forrest', 'Gainesville', 'Hepburn Springs', 'Huskisson',
       'Inverloch', 'Lakes Entrance', 'Lakeview', 'Lincoln', 'Lorn',
       'Macedon', 'Macon', 'Mayfield', 'Mc Millan', 'Middleton Beach',
       'Miller', 'Monroe', 'Montville', 'Ojo Caliente', 'Orlando',
       'Palm Cove', 'Paynesville', 'Penola', 'Pensacola',
       'Phillip Island', 'Pocatello', 'Potrero', 'Princeton',
       'Rest of Hawaii', 'Savannah', 'Singapore', 'Sioux City',
       'Tampa Bay', 'Tanunda', 'Trentham East

In [299]:
df['City'].nunique()

141

In [301]:
df['City'].value_counts()

City
New Delhi           5473
Gurgaon             1118
Noida               1080
Faridabad            251
Ghaziabad             25
                    ... 
Panchkula              1
Mc Millan              1
Mayfield               1
Macedon                1
Vineland Station       1
Name: count, Length: 141, dtype: int64

Comment:
- New Delhi has most of the restaurants.
- There are 141 unique cities.

## Cuisines

In [305]:
df['Cuisines'].unique()

array(['French, Japanese, Desserts', 'Japanese',
       'Seafood, Asian, Filipino, Indian', ..., 'Burger, Izgara',
       'World Cuisine, Patisserie, Cafe', 'Italian, World Cuisine'],
      dtype=object)

In [307]:
df[['Restaurant Name', 'Cuisines']]

Restaurant Name                          Cuisines
0            Le Petit Souffle        French, Japanese, Desserts
1            Izakaya Kikufuji                          Japanese
2        Heat  Edsa ShangriLa  Seafood, Asian, Filipino, Indian
3                        Ooma                   Japanese, Sushi
4                 Sambo Kojin                  Japanese, Korean
...                       ...                               ...
9546               Naml Gurme                           Turkish
9547                Ceviz Aac   World Cuisine, Patisserie, Cafe
9548                    Huqqa            Italian, World Cuisine
9549                 Ak Kahve                   Restaurant Cafe
9550  Walters Coffee Roastery                              Cafe

[9551 rows x 2 columns]

In [309]:
df = df.drop_duplicates(subset='Cuisines')

In [311]:
df['Cuisines'] = df['Cuisines'].str.split(', ')
df= df.explode('Cuisines')
df.head()

Restaurant ID       Restaurant Name  Country Code              City  \
0        6317637      Le Petit Souffle           162       Makati City   
0        6317637      Le Petit Souffle           162       Makati City   
0        6317637      Le Petit Souffle           162       Makati City   
1        6304287      Izakaya Kikufuji           162       Makati City   
2        6300002  Heat  Edsa ShangriLa           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
0  Third Floor, Century City Mall, Kalayaan Avenu...   
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa ShangriLa, 1 Garden Way, Ortigas, Mandalu...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
0   Century City Mall, Poblacion, Makati City   
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2   Edsa ShangriLa, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa ShangriLa, Ortigas, Mandaluyong City, Man...  121.056831  14.581404   

   Cuisines  ...        Currency Has Table booking Has Online delivery  \
0    French  ...  Botswana PulaP               Yes                  No   
0  Japanese  ...  Botswana PulaP               Yes                  No   
0  Desserts  ...  Botswana PulaP               Yes                  No   
1  Japanese  ...  Botswana PulaP               Yes                  No   
2   Seafood  ...  Botswana PulaP               Yes                  No   

  Is delivering now Switch to order menu Price range  Aggregate rating  \
0                No                   No           3               4.8   
0                No                   No           3               4.8   
0                No                   No           3               4.8   
1                No                   No           3               4.5   
2                No                   No           4               4.4   

   Rating color Rating text Votes  
0    Dark Green   Excellent   314  
0    Dark Green   Excellent   314  
0    Dark Green   Excellent   314  
1    Dark Green   Excellent   591  
2         Green   Very Good   270  

[5 rows x 21 columns]

In [313]:
df['Cuisines'].unique()

array(['French', 'Japanese', 'Desserts', 'Seafood', 'Asian', 'Filipino',
       'Indian', 'Sushi', 'Korean', 'Chinese', 'European', 'Mexican',
       'American', 'Ice Cream', 'Cafe', 'Italian', 'Pizza', 'Bakery',
       'Mediterranean', 'Fast Food', 'Brazilian', 'Arabian', 'Bar Food',
       'Grill', 'International', 'Peruvian', 'Latin American', 'Burger',
       'Juices', 'Healthy Food', 'Beverages', 'Lebanese', 'Sandwich',
       'Steak', 'BBQ', 'Gourmet Fast Food', 'Mineira', 'North Eastern',
       nan, 'Coffee and Tea', 'Vegetarian', 'Tapas', 'Breakfast', 'Diner',
       'Southern', 'Southwestern', 'Spanish', 'Argentine', 'Caribbean',
       'German', 'Vietnamese', 'Thai', 'Modern Australian', 'Teriyaki',
       'Cajun', 'Canadian', 'TexMex', 'Middle Eastern', 'Greek',
       'Bubble Tea', 'Tea', 'Australian', 'Fusion', 'Cuban', 'Hawaiian',
       'Salad', 'Irish', 'New American', 'Soul Food', 'Turkish',
       'Pub Food', 'Persian', 'Continental', 'Singaporean', 'Malay',
       '

In [315]:
df['Cuisines'].value_counts()

Cuisines
North Indian      685
Chinese           556
Italian           398
Continental       382
Fast Food         297
                 ... 
Drinks Only         1
Malay               1
Iranian             1
South American      1
Brek                1
Name: count, Length: 145, dtype: int64

In [317]:
df['Cuisines'].nunique()

145

Comment:
- Most demanded cuisines are North Indian and Chinese.
- There are 145 unique cuisines.
- Multicuisines restaurants were split to count each cuisines.

**Top three cuisinies**

In [321]:
df_top3cuisines=df['Cuisines'].value_counts().reset_index()
df_top3cuisines.columns=['Cuisines','Restaurant Count']
df_top3cuisines.head(3)

Cuisines  Restaurant Count
0  North Indian               685
1       Chinese               556
2       Italian               398

**Top three cities with highest number of restaurant**

In [324]:
df_top3restaurant=df['City'].value_counts().reset_index()
df_top3restaurant.columns=['City','Count']
df_top3restaurant.head(3)

City  Count
0  New Delhi   2197
1    Gurgaon    831
2      Noida    265